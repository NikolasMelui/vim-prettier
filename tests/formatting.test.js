const HeadlessRemoteClient = require('vim-driver/dist/HeadlessRemoteClient');
const Server = require('vim-driver/dist/Server');

const HOST = '127.0.0.1';
const PORT = 1337;

let server;
let remote;

beforeAll(async () => {
  server = new Server();
  await server.listen(HOST, PORT);
});

afterAll(async () => {
  await server.close();
});

beforeEach(async () => {
  remote = new HeadlessRemoteClient({host: HOST, port: PORT});
  await remote.connect(server);
});

afterEach(async () => {
  if (remote.isConnected()) {
    await remote.close();
  }
});

test('PrettierVersion returns pluggin version', async () => {
  const result = await remote.execute('PrettierVersion');
  expect(result).toMatchSnapshot();
});
