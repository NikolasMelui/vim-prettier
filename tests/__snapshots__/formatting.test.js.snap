// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrettierVersion returns pluggin version 1`] = `"0.2.7"`;
